test/bin/manage_webserver.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+#
+# This script should be run on the hypervisor to set up an nginx
+# file-server for the images used by VMs running test scenarios.
+
+set -euo pipefail
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+# shellcheck source=test/bin/common.sh
+source "${SCRIPTDIR}/common.sh"
+
+usage() {
+    cat - <<EOF
+${BASH_SOURCE[0]} (start|stop)
+
+  -h           Show this help.
+
+start: Start the nginx web server. 
+
+stop: Stop the nginx web server.
+
+EOF
+}
+
+nginx_conf="
+worker_processes 32;
+events {
+}
+http {
+    access_log /dev/null;
+    error_log  ${IMAGEDIR}/nginx_error.log;
+    server {
+        listen ${WEB_SERVER_PORT};
+        listen [::]:${WEB_SERVER_PORT};
+        root   ${IMAGEDIR};
+        autoindex on;
+    }
+
+    # Timeout during which a keep-alive client connection will stay open on the server
+    # Default: 75s
+    keepalive_timeout 300s;
+
+    # Timeout for transmitting a response to the client
+    # Default: 60s
+    send_timeout 300s;
+
+    # Buffers used for reading response from a disk
+    # Default: 2 32k
+    output_buffers 2 1m;
+}
+pid ${IMAGEDIR}/nginx.pid;
+daemon on;
+"
+
+action_stop() {
+    echo "Stopping web server"
+    sudo pkill nginx || true
+    exit 0
+}
+
+action_start() {
+    echo "Starting web server in ${IMAGEDIR}"
+    mkdir -p "${IMAGEDIR}"
+    cd "${IMAGEDIR}"
+
+    NGINX_CONFIG="${IMAGEDIR}/nginx.conf"
+    # See the https://nginx.org/en/docs/http/ngx_http_core_module.html page for
+    # a full list of HTTP configuration directives
+    echo "${nginx_conf}" > "${NGINX_CONFIG}"
```

Comment:
If you're using echo to create a file, I think all the formatting will go away.
Let's use in-place editing?

Reply:
You can separate it into a function if you'd like to simplify the current function code.

Reply:
I think it keeps the formatting, as long as you're using double quotes around the variable: `"${nginx_conf}"`

Reply:
Changed to be more consistent